tdrs-backend/tdpservice/parsers/case_consistency_validator.py

```diff
@@ -183,76 +210,60 @@ def __validate_s1_records_are_related(self):
-                            record,
-                            field='RPT_MONTH_YEAR',
-                            msg=(
-                                f'Every {t1_model_name} record should have at least one '
-                                f'corresponding {t2_model_name} or {t3_model_name} record '
-                                f'with the same RPT_MONTH_YEAR and CASE_NUMBER.'
-                            )
-                        )
-                        num_errors += 1
-
-                else:
-                    # loop through all t2s and t3s
-                    # to find record where FAMILY_AFFILIATION == 1
-                    num_errors += self.__validate_family_affiliation(num_errors, t1s, t2s, t3s, (
-                            f'Every {t1_model_name} record should have at least one corresponding '
-                            f'{t2_model_name} or {t3_model_name} record with the same RPT_MONTH_YEAR and '
-                            f'CASE_NUMBER, where FAMILY_AFFILIATION==1'
-                        ))
-
-                    # the successful route
-                    # pass
-            else:
-                for record, schema in t2s:
+        t1s = self.sorted_cases.get(t1_model, [])
+        t2s = self.sorted_cases.get(t2_model, [])
+        t3s = self.sorted_cases.get(t3_model, [])
+
+        if len(t1s) > 0:
+            if len(t2s) == 0 and len(t3s) == 0:
+                for record, schema in t1s:
                     self.__generate_and_add_error(
                         schema,
                         record,
                         field='RPT_MONTH_YEAR',
                         msg=(
-                            f'Every {t2_model_name} record should have at least one corresponding '
-                            f'{t1_model_name} record with the same RPT_MONTH_YEAR and CASE_NUMBER.'
+                            f'Every {t1_model_name} record should have at least one '
+                            f'corresponding {t2_model_name} or {t3_model_name} record '
+                            f'with the same RPT_MONTH_YEAR and CASE_NUMBER.'
                         )
                     )
                     num_errors += 1
 
-                for record, schema in t3s:
-                    self.__generate_and_add_error(
-                        schema,
-                        record,
-                        field='RPT_MONTH_YEAR',
-                        msg=(
-                            f'Every {t3_model_name} record should have at least one corresponding '
-                            f'{t1_model_name} record with the same RPT_MONTH_YEAR and CASE_NUMBER.'
-                        )
+            else:
+                # loop through all t2s and t3s
+                # to find record where FAMILY_AFFILIATION == 1
+                num_errors += self.__validate_family_affiliation(num_errors, t1s, t2s, t3s, (
+                        f'Every {t1_model_name} record should have at least one corresponding '
```

Comment:
duplicative error messages produced by the following file

[SSP_Sec1.txt](https://github.com/user-attachments/files/15990485/SSP_Sec1.txt)


```
Duplicate record detected with record type M1 at line 3. Record is a duplicate of the record at line number 2.
Every M1 record should have at least one corresponding M2 or M3 record with the same RPT_MONTH_YEAR and CASE_NUMBER.
Every M1 record should have at least one corresponding M2 or M3 record with the same RPT_MONTH_YEAR and CASE_NUMBER.
```